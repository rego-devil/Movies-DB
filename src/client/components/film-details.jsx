import React from 'react';
import { Link } from 'react-router-dom';
import Loading  from './loading';

export class FilmDetails extends React.Component {

  componentWillMount() {
    this.props.getFilm(this.props.match.params.id);
  }

  render() {
    const {currentFilmDetails, onClearCurrentFilm, isFetching} = this.props;    

    if(!currentFilmDetails) 
      return null

    return (
      <div className="filmDetails">
       
        <img className="filmDetails__img"  src={currentFilmDetails.poster_path} />
        <div className="filmDetails__overview">
          <div className="filmDetails__top">
            <div className="filmDetails__title">{currentFilmDetails.title}</div>
            <div className="filmDetails__rating">{currentFilmDetails.vote_average}</div>
            
              <Link to="/">
                <button className="filmDetails__backToSearch" onClick={() => onClearCurrentFilm(null)}>
                  Back to search
                  </button>
              </Link>
            
          </div>
          <div className="filmDetails__middle">
            <div className="filmDetails__date">{currentFilmDetails.release_date && currentFilmDetails.release_date.split('-')[0]}</div>
            <div className="filmDetails__duration">{currentFilmDetails.runtime ? currentFilmDetails.runtime + ` min.` : ''}</div>
          </div>
          <div className="filmDetails__description">{currentFilmDetails.overview}</div>
          <div className="filmDetails__genres">
          {
            currentFilmDetails.genres && currentFilmDetails.genres.map((item, i) => (i > 0 ? ', ':'') + item)
          }
          </div>
        </div>
      </div>
      
      
      // currentFilmDetails ? (
      //     result
      // ) : (isFetching ? 
      //   <React.Fragment>
      //     { console.log('----3----')}
      //     <Loading />
      //   </React.Fragment> : '')
        
    )
  }
  
}