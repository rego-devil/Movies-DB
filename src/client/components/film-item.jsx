import React from 'react';
import propTypes from 'prop-types';


export const FilmItem = ({title, genres, release_date, poster_path}) => {

	let genreList = genres.map((item,i) => {
		return (i > 0 ? ', ':'') + item;
	});

	return (
		<React.Fragment>
			<img src={poster_path} className="filmItem__img" />
			<div className="filmItem__description">
					<div className="filmItem__title">{title}</div>
					<div className="filmItem__releaseYear">{release_date.split('-')[0]}</div>
			</div>
			<div className="filmItem__genre">{genreList}</div>
		</React.Fragment>
	)
}

FilmItem.propTypes = {
	title: propTypes.string.isRequired,
	release_date: propTypes.string.isRequired,
	poster_path: propTypes.string.isRequired,
	genres: propTypes.array,
}