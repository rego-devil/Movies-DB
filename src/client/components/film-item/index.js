export { default } from './film-item';
