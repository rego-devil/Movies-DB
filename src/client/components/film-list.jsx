import React from 'react';
import {FilmItem, FilmSort} from './';

export class FilmList extends React.Component {
	constructor() {
		super();
	}

	render() {
		const {filmList, onShowFilmDetails, onSelectFilm} = this.props;		
		return (
			filmList.length > 0 ? (
				<React.Fragment>
				<FilmSort />
				<ul className="filmList">
					{
						filmList.map((item) => 
							<li key={item.id} className="filmItem" onClick={() =>  onShowFilmDetails(item) }>
								<FilmItem {...item} />
							</li>
						)
					}
				</ul>
				</React.Fragment>
			) : (
				<div className="noFilms">No films found</div>
			)
		)
	}
}


