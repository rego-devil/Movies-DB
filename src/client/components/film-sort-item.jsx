import React from 'react';


export const SortItem = ({sortBy, onSort, currentSortField, value, sortOrderLocal }) => (
	<span 
		className={`filmSort__item ${currentSortField === sortBy ? 'filmSort__item--active' : ''}`} 
		onClick={() => onSort({sortBy: currentSortField, sortOrder: sortOrderLocal}) } 
	>
		{value}
		{currentSortField === sortBy ? sortOrderLocal === 'asc' ? String.fromCharCode(8593) : String.fromCharCode(8595) : ''  }
	</span>
)