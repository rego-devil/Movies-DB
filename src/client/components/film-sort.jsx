import React from 'react';

export const FilmSort = () => (
  <div className="filmInfo">
		<div className="filmInfo__count">7 film found</div>
		<div className="filmInfo__sortBy">Sort by: 
			<span className="filmInfo__sortByActive">release date</span>
			<span>rating</span>
		</div>
	</div>
)