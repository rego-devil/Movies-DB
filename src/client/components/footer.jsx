import React from 'react';


export const Footer = () => (
  <div className="footer">
      <div className="wrapper">
        	Project Name
      </div>
  </div>
)