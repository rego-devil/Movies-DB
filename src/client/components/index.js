export { Header } from './header';
export { FilmItem } from './film-item';
export { FilmList } from './film-list';
export { FilmDetails } from './film-details';
export { FilmSort } from './film-sort';
export { SortItem } from './film-sort-item';
export { Search } from './search';
export { Footer } from './footer';
