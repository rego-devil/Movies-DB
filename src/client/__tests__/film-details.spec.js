import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { FilmDetails } from '../components/film-details';

describe('film Item component', () => {
  let renderedComponent;
  const minProps = {
    title: 'default',
    genres: [],
    release_date: '1939-08-15',
    vote_average: 8,
    poster_path: 'ulrToImg',
    runtime: '',
  };

  test('+++ Match snapshot', () => {
    const renderedComponent = shallow(<FilmDetails {...minProps} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const renderedComponent = shallow(<FilmDetails {...minProps} />);
    expect(renderedComponent.length).toEqual(1);

    // console.log(renderedComponent.debug());
  });

  test('Check runtime condition', () => {
    let renderedComponent,
      runtime;

    runtime = 128;
    renderedComponent = shallow(<FilmDetails {...minProps} runtime={runtime} />);
    expect(renderedComponent.find('.filmDetails__duration').text()).toEqual(`${runtime} min.`);

    runtime = '';
    renderedComponent = shallow(<FilmDetails {...minProps} runtime={runtime} />);
    expect(renderedComponent.find('.filmDetails__duration').text()).toEqual(runtime);

    // console.log(renderedComponent.debug());
  });

  // test('click event', () => {
  //     const wrapper = shallow(<FilmItem  />);
  //     // wrapper.find('.filmItem').simulate('click');
  // });
});
