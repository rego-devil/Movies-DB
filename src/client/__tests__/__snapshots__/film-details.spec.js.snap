// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`film Item component +++ Match snapshot 1`] = `
<div
  className="filmDetails"
>
  <img
    className="filmDetails__img"
    src="ulrToImg"
  />
  <div
    className="filmDetails__overview"
  >
    <div
      className="filmDetails__top"
    >
      <div
        className="filmDetails__title"
      >
        default
      </div>
      <div
        className="filmDetails__rating"
      >
        8
      </div>
      <button
        className="filmDetails__backToSearch"
        onClick={[Function]}
      >
        Back to search
      </button>
    </div>
    <div
      className="filmDetails__middle"
    >
      <div
        className="filmDetails__date"
      >
        1939
      </div>
      <div
        className="filmDetails__duration"
      />
    </div>
    <div
      className="filmDetails__description"
    />
    <div
      className="filmDetails__genres"
    />
  </div>
</div>
`;
